src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -48,13 +49,13 @@ public ViewBuffer(IViewBufferScope bufferScope, string name, int pageSize)
             _name = name;
             _pageSize = pageSize;
 
-            Pages = new List<ViewBufferPage>();
+            _pages = new ViewBufferPages();
         }
 
         /// <summary>
         /// Gets the backing buffer.
         /// </summary>
-        public IList<ViewBufferPage> Pages { get; }
+        public IReadOnlyList<ViewBufferPage> Pages => _pages;
```

Comment:
The `ViewBuffer` / `ViewBufferPages` merge won't allow to directly return a member. I will have to create a intermediate List/Array. Can I still let the `Pages` property, or is it clearer to rename it as `GetPages()` method?